sdks/java/io/kafka/src/test/java/org/apache/beam/sdk/io/kafka/KafkaIOTest.java

```diff
@@ -723,6 +724,31 @@ public void testUnboundedSourceWithExplicitPartitions() {
     p.run();
   }
 
+  @Test
```

Comment:
can you add a test, or update this test, such that you don't match all the topics?

Reply:
Added one for partial matches and one for no matches.